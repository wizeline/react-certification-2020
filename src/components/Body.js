import styled from 'styled-components'


export const Container = styled.div`
border: 10px ;
display: static;
justify-content: center;
margin-top: 0px;
`;

    export const Title = styled.h1`
        
  
    font-family: sans-serif;    
    @media (min-width: 600px){
      margin-bottom: 25px;
      margin-top: 10px;
    margin-left: auto;
    margin-right: auto;
    font-size: 70px;
    justify-content: center;
    text-align-last: center;
    }
        
    `;
    export const Vid = styled.div`
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    margin-top: 60px;
    margin-left: auto;
    margin-right: auto;
    opacity: 10;
 
    `;

    export const VideoItems = styled.div`
    flex: ${props => props.ratio};
    height: 400px;
  border-radius: 5px;
  border: 1px solid #ddd;
  margin: 10px;
  flex-basis: calc(20% - 22px);
  cursor: pointer;
  padding: 9px 10px;
    `;

    export const Parag = styled.div`
    margin-top: 70px;
  font-size: 18px;
  white-space: pre-line;
  text-align: left;
  font-family: sans-serif;
  word-break: break-all;
  white-space: pre-line;
  text-align-last: left;
}
  
    `;

    export const ImgBox = styled.div`
    display: block;
    
   width: 300px;
    height: 100px;
    
    `;
    export const IMG = styled.img`
    display: block;
    margin-left: auto;
    margin-right: auto;
    max-width:100%;
    background-size: contain;  
    `;
    export const TitleVideo = styled.p`
    font-size: 20px;
  font-weight: bold;
  padding: 10px;
  text-align-last: left;
  }
  
    `;

    export const Secondary = styled.p`
    font-size: 16px;
  overflow:hidden;
  padding: px;
  
  @media (min-width: 600px){
  
  }
    `;
    
   
