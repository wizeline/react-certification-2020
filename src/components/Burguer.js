import React, { useState } from 'react';
import styled from 'styled-components';
import LeftNav from './LeftNave';

const StyledBurger = styled.div`
  width: 2rem;
  height: 2rem;
  position: fixed;
  top: 15px;
  left: 20px;
  z-index: 100;
  display: none;
  @media (min-width: 320px) {
    display: flex;
    justify-content: space-around;
    flex-flow: column nowrap;
  }
  div {
    width: 2rem;
    height: 0.25rem;
    background-color: ${({ open }) => open ? '#ccc' : '#333'};
    border-radius: 10px;
    transform-origin: 1px;
    transition: all 0.3s linear;
    z-index: 100;
    &:nth-child(1) {
      transform: ${({ open }) => open ? 'rotate(45deg)' : 'rotate(0)'};
    }
    &:nth-child(2) {
      transform: ${({ open }) => open ? 'translateX(100%)' : 'translateX(50)'};
      opacity: ${({ open }) => open ? 0 : 1};
    }
    &:nth-child(3) {
      transform: ${({ open }) => open ? 'rotate(-45deg)' : 'rotate(0)'};
    }
  }
`;

const Burger = () => {
  const [open, setOpen] = useState(false)
  
  return (
    <>
      <StyledBurger open={open} onClick={() => setOpen(!open)}>
        <div />
        <div />
        <div />
      </StyledBurger>
      <LeftNav open={open} closeBurguer={() => setOpen(!open)}/>
    </>
  )
};

export default Burger;