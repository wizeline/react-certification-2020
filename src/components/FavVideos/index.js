export { default } from './FavVideos.component';