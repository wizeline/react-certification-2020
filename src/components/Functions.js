import React from 'react'
export default function checked(){
    if(document.getElementById("check").checked = false){
      console.log("entro");
    }
  }