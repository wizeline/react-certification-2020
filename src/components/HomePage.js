import styled from 'styled-components'

export const Container = styled.body`
padding: 0;
    margin: 0;
  
   
    font-family: sans-serif;
    
`;