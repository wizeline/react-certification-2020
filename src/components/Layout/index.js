import { useQueryContext } from "./Layout.component";

export { default } from "./Layout.component";
export { useQueryContext };
