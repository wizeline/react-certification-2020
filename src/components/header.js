import styled from 'styled-components'

export const Head = styled.head`
color: white;
margin: 0;
height: auto;

  
@media (min-width: 600px){
      flex-direction: row ;
      overflow: hidden;
      display: block;
      width: 100%;
      position: static;
    }
`;

export const Input = styled.input`
    float:left;
    z-index: -1px; 
    alignItems: "stretch",
    background-image: url('images/microphone.png'),
    background-repeat: no-repeat,
    background-position: right,
    display: "absolute",
    flex-wrap: wrap;
    border: none;
    background-color: #074b3d;
    display: flex;
    size: normal;
    align-items: center;
    justify-content: center;
    border-radius: 5px;
    padding: .75em ;
    margin: 1em;  
    color: white;
    opacity: 0.5;
    @media (min-width: 600px){
      width: 15%;
      flex-direction: row ;
    }
    ::placeholder {
      color: white;
    `;

  
   
    

    

    
