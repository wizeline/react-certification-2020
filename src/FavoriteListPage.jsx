import React, {useState,useContext} from 'react';
import {Container,  Vid} from './components/Body';
//import {VideoItems, Parag, TitleVideo, Secondary, IMG} from './components/VideoPlayer.js';
import SearchContext from './GlobalVars/SearchContext';
import VideoItem from './VideoItem';

function FavoriteListPage(){
   
  const { setTitles, setVideoList } = useContext(SearchContext);

    return(
        <div>
            <Container >
            <Vid >
                  {Object.entries(localStorage).map(([key,value]) => (  

                    (key !="session" ) ?
                      <VideoItem setTitles={setTitles}
                      title={JSON.parse(value).title}
                      description={JSON.parse(value).description}
                      url={JSON.parse(value).url}
                      videoId={JSON.parse(value).videoId}></VideoItem>:null
                    
                  ))}  
                </Vid>
                
                  
            </Container>
            
            </div>
    );
}

export default FavoriteListPage;