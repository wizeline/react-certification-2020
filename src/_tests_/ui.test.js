import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';
import ReactDOM from 'react-dom';
import {Title, Container} from '../components/Body';
import { MemoryRouter } from 'react-router';
//import {render, fireEvent, cleanup} from '@testing-library/react';
import Main from '../main';
import FavoriteListPage from '../FavoriteListPage';
import { mount, configure } from 'enzyme';
import Circle from '../components/menu';




describe('Correct style in Body components', () => {
    test('Title : Correct style', () => {
        const wrapper =renderer.create(<Title/>).toJSON();
        expect(wrapper).toHaveStyleRule('font-family', 'sans-serif');
        //expect(wrapper).toHaveStyleRule('color', 'black');
    });
    test('Container: Corect Styles', () => {
        const wrapper = renderer.create(<Container />).toJSON();
       expect(wrapper).toHaveStyleRule('display', 'static');
        expect(wrapper).toHaveStyleRule('justify-content', 'center');
        expect(wrapper).toHaveStyleRule('margin-top', '0px')
    })
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Container />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// it('Text in state is changed when button clicked', () => {
//   const { getByText } = render(<Input />);

//   expect(getByText(/Initial/i).textContent).toBe("Initial State")

//   fireEvent.click(getByText("State Change Button"))

//   expect(getByText(/Initial/i).textContent).toBe("Initial State Changed")
// })

test('invalid path should redirect to 404', () => {
  configure({ adapter: new Adapter() })
  const wrapper= mount(
    <MemoryRouter initialEntries={[ '/favorite' ]}>
      <Main/>
    </MemoryRouter>
  )
  expect(wrapper.find(FavoriteListPage)).toHaveLength(0);
  
});

it('render correctly text component', () => {  
  const Menu = renderer.create(<Circle />).toJSON();
  expect(Menu).toMatchSnapshot();
});





