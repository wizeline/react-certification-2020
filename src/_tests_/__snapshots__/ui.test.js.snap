// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly text component 1`] = `
.c1 {
  border-radius: 999999px;
  height: 45px;
  z-index: -1;
  background: transparent;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: .5rem .5rem;
  float: left;
  padding: .5em;
  margin: .4em;
}

.c3 {
  width: 20px;
  height: 3px;
  background: white;
  margin: 3px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0:hover .c2 {
  background-color: #536162;
  opacity: .7;
}

@media (min-width:600px) {
  .c1 {
    width: 5%;
  }
}

<div
  className="c0 c1"
  onClick={[Function]}
  type="button"
>
  <i>
    <div
      className="c2 c3"
    />
    <div
      className="c2 c3"
    />
    <div
      className="c2 c3"
    />
  </i>
</div>
`;
